src/constants.lua

```diff
@@ -1,13 +1,19 @@
+--[[
+	NOTE: constants is used throughout the codebase, so avoid imports of any modules in this file to prevent circular dependencies
+]]
+--
 local constants = {}
 
+--- CONVERSION HELPERS
+constants.DENOMINATION = 6
 -- @alias mARIO number
 -- intentionally not exposed so all callers use ARIOToMARIO for consistency
-local mARIO_PER_ARIO = 1000000
+local mARIO_PER_ARIO = 10 ^ constants.DENOMINATION -- 1 million mARIO per ARIO
 
 --- @param ARIO number
 --- @return mARIO mARIO the amount of mario for the given ARIO
 function constants.ARIOToMARIO(ARIO)
-	return ARIO * mARIO_PER_ARIO
+	return math.floor(ARIO * mARIO_PER_ARIO)
```

Comment:
Did the absence of floor reveal an issue somewhere? It makes sense that we want to never have fractional mARIO but I could only see that happening if an ARIO value has too many decimals (which I suppose we end up having to guard for somewhere).

Reply:
No - it just showed the decimal in error messages - implying there could be fractional mARIO provided for quantities